examples/passport-oauth2-login/src/authentication-strategies/oauth2-strategy.ts

```diff
@@ -0,0 +1,104 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: example-passport-oauth2-login
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  asAuthStrategy,
+  AuthenticationStrategy,
+  UserProfileFactory,
+} from '@loopback/authentication';
+import {StrategyAdapter} from './strategy-adapter';
+import {Strategy, StrategyOption, Profile} from 'passport-facebook';
+import {bind, inject} from '@loopback/context';
+import {Request} from '@loopback/rest';
+import {UserProfile, securityId} from '@loopback/security';
+import {UserService, UserServiceBindings} from '../services';
+import {User} from '../models';
+import {asSpecEnhancer, OASEnhancer} from '@loopback/openapi-v3';
+
+const options: StrategyOption = {
```

Comment:
Are these real ids? Perhaps remove them?

Reply:
@emonddr , yep I created a test app in facebook, it is not a security risk as of now. but I am going to remove it shortl